### DISCUSSION TITLE
Custom properties and Auto-create customer

### DISCUSSION THREAD

**Discussion by HappyPathAB** [Labs64/NetLicensing-Community#82] [Ideas & Feature Requests]
I am using the C# API and have a product set to auto-create customer.
If I include [custom properties](https://netlicensing.io/wiki/custom-properties) in the initial `validate()` call which creates the customer, they don't seem to be honored. They're not returned in the response, nor in subsequent validation responses.

Is it not possible to include custom properties in this call?
Or am I doing something wrong?

**Comment by r-brown** [comment]
Hello @HappyPathAB ,
As of now [LicenseeServices.validate()](https://netlicensing.io/wiki/licensee-services#validate-licensee) doesn't set custom properties upon validation.

[LicenseeServices.update()](https://netlicensing.io/wiki/licensee-services#update-licensee) operation can be used to update `Liecnsee` entity and set additional custom properties.
Required [API Key role](https://netlicensing.io/wiki/security#api-key-identification) for this operation: `ROLE_APIKEY_OPERATION`

**Comment by r-brown** [comment]
New Feature Request issue added to the Core backlog.

**Comment by HappyPathAB** [comment]
So this means in effect that custom properties can't be set when using auto-create customer -- because you don't want to include an Operation-level API key in code shipped to clients.

Is my understanding correct that this is a limitation on the server side and not just in the C# API?
And one that you plan to address?

Sorry if this seems dense, but there's no "Core" repository on the Labs64 GitHub page so I'm not sure what your feature request comment means.

**Comment by r-brown** [comment]
Hi @HappyPathAB sorry for the confusion - I was referring to the _NetLicensing Core services_ (serving RESTful API).
And you are totally right - this is not necessarly related to one of the NetLicensing [client libraries](https://netlicensing.io/wiki/client-libraries), rather the change need to be implemented in API ( _Core :)_ ).

Until this will be addressed, the above workaround with the _LicenseeServices.update()_ can be used.

**Comment by r-brown** [comment]
NetLicensing v2.8.16 introduced specifying of the custom properties for new customers on [validate call](https://netlicensing.io/wiki/licensee-services#validate-licensee).

Example:
```shell
curl -X POST --header 'Content-Type: application/x-www-form-urlencoded' \
  --header 'Accept: application/xml' \
  --header 'Authorization: Basic <obscured>' \
  -d 'productNumber=P93DQU34K&param1=value1&param2=value2' 'https://go.netlicensing.io/core/v2/rest/licensee/CUST-T03/validate'
```
... results in the customer properties / tags set in the customer profile:
<img width="841" alt="netlicensing-validate-tags" src="https://user-images.githubusercontent.com/1361258/204131724-d0236073-83bc-4a26-9784-6c625e63e739.png">
